src/lib/picker/category.component.ts

```diff
@@ -126,12 +161,36 @@ export class CategoryComponent implements OnInit {
       // this.labelSpanStyles = { position: 'absolute' };
     }
   }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.emojis?.currentValue?.length !== changes.emojis?.previousValue?.length) {
+      this.ngAfterViewInit();
+    }
+  }
+
+  ngAfterViewInit() {
+    if (!this.virtualize || !this.emojis?.length) {
+      return;
+    }
+
+    const parent = this.container.nativeElement.parentNode.parentNode;
+    const { width } = parent.getBoundingClientRect();
+
+    const rows = Math.ceil((this.emojis.length * (this.emojiSize + 12)) / width);
```

Comment:
might make sense to get per row first, since you aren't rendering 9.3 per row

```ts
const perRow = Math.floor(width / (this.emojiSize + 12));
const rows = Math.ceil(this.emojis.length / perRow);
```

Reply:
That's a good idea. Ideally, it should render per row anyway, calculating the before and after placeholder sizes for the larger categories. I know how to do it, but just looking for some time to get that down 👍 . Thank you!